src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -313,13 +325,8 @@ void Instance::HandleSetCookingParameters(HandlerContext & ctx, const Commands::
             cookMode.HasValue() || cookTimeSec.HasValue() || wattSettingIndex.HasValue(), status = Status::InvalidCommand;
             ChipLogError(Zcl, "Microwave Oven Control: Failed to set cooking parameters, all command fields are missing "));
 
-        // count of supported watt levels must greater than 0
-        VerifyOrExit(
-            mSupportedWattLevels > 0,
-            ChipLogError(Zcl, "Microwave Oven Control: Failed to set wattSettingIndex, count of supported watt levels is 0"));
-
-        reqWattSettingIndex = wattSettingIndex.ValueOr(mSupportedWattLevels - 1);
```

Comment:
So in the old code, since we had checked mSupportedWattLevels > 0, this was safe; we just needed to explicitly cast the subtraction result to `uint8_t`, right?

Why are we adding all the other complexity?

Reply:
I see, we will change to add the static_cast on `mSupportedWattLevels - 1`